### PR TITLE
[#7636] Change naming scheme used by feature test macros (main)

### PR DESCRIPTION
**Pull request by korydraughn**
The `IRODS_LIBRARY_FEATURE_` prefix is fine/good. That's the namespace for the macros.

I'm not a big fan of the `_LIBRARY_`, `_API_ENDPOINT_` component though.

Is there a better naming scheme for those?
Would a shorter name be better? i.e. `_LIB_`, `_API_`
Does it make since to continue namespacing the macros after the `IRODS_LIBRARY_FEATURE_` prefix?
Would the names be better if removed?

I think I can reuse the documentation for the original macros. Will investigate.

### CODE REVIEW

**Comment by SwooshyCueb** [comment]
> I'm not a big fan of the `_LIBRARY_`, `_API_ENDPOINT_` component though.

I'm not a fan of `_LIBRARY_` either.

> Is there a better naming scheme for those? 
> Would a shorter name be better? i.e. `_LIB_`, `_API_` 

I like both of these. However...

> Does it make since to continue namespacing the macros after the `IRODS_LIBRARY_FEATURE_` prefix?
> Would the names be better if removed?

I think removing the namespacing/prefixing past `IRODS_LIBRARY_FEATURE_` makes sense. Since we're using numbers that can be incremented, I don't think there is as much need for that kind of granularity. Should we ever need to differentiate, that can be done on a case-by-case basis.

**Comment by korydraughn** [comment]
Updated.

Need to render the docs to see how things look now.

File: lib/core/include/irods/library_features.h [hunk 1]
```diff
@@ -21,42 +21,84 @@
...
+/// \sa Prefer IRODS_LIBRARY_FEATURE_LIB_TICKET_ADMINISTRATION
+///
 /// \since 4.3.1
-#define IRODS_HAS_LIBRARY_TICKET_ADMINISTRATION                              202209L
+#define IRODS_HAS_LIBRARY_TICKET_ADMINISTRATION                               202209L
+
+/// An alias for IRODS_HAS_LIBRARY_TICKET_ADMINISTRATION.
+///
+/// \since 4.3.2
+#define IRODS_LIBRARY_FEATURE_LIB_TICKET_ADMINISTRATION                       202209L
```
Comment (alanking) [inline lib/core/include/irods/library_features.h:32]: Would it make sense to have the alias macros defined by the other macro? In other words...
```c
#define IRODS_LIBRARY_FEATURE_LIB_TICKET_ADMINISTRATION                       202209L
#define IRODS_HAS_LIBRARY_TICKET_ADMINISTRATION                               IRODS_LIBRARY_FEATURE_LIB_TICKET_ADMINISTRATION
```
- OR -
```c
#define IRODS_HAS_LIBRARY_TICKET_ADMINISTRATION                               202209L
#define IRODS_LIBRARY_FEATURE_LIB_TICKET_ADMINISTRATION                       IRODS_HAS_LIBRARY_TICKET_ADMINISTRATION
```

**Review by alanking — COMMENTED** [review]
Just had one minor suggestion

File: lib/core/include/irods/library_features.h (hunk 1 above)
Comment (korydraughn) [reply lib/core/include/irods/library_features.h:32]: I considered that and decided to keep them separate in case they end up diverging (on purpose).

Comment (korydraughn) [reply lib/core/include/irods/library_features.h:32]: I'm going to make them aliases.

It just makes more sense from a maintainability perspective.

Comment (korydraughn) [reply lib/core/include/irods/library_features.h]: Done.

**Review by SwooshyCueb — COMMENTED** [review]
Was there further discussion about namespacing/prefixing past `IRODS_LIBRARY_FEATURE_` that I missed? I see `_LIB_` and `_API_`, but the last discussion about it in this PR was @korydraughn agreeing with my statement that removing the added prefixing/namespacing makes sense.

**Comment by korydraughn** [comment]
There wasn't really any additional conversation following ours. @alanking made a suggestion about macro aliasing.

Did I misinterpret what you meant in regard to `_LIB_` and `_API_`?

**Comment by korydraughn** [comment]
After some discussion, the decision is to remove the internal namespacing (i.e. `_LIB_`, `_API_`) and only keep the `IRODS_LIBRARY_FEATURE_` prefix.

**Comment by korydraughn** [comment]
Internal/Nested namespaces removed.

Running the unit test.

**Comment by korydraughn** [comment]
The library features unit test passed.

Squashing now.

**Comment by korydraughn** [comment]
Squashed.

**Review by SwooshyCueb — APPROVED** [review]

**Comment by korydraughn** [comment]
#'d

**Pull request merged**